pandas/core/resample.py

```diff
@@ -853,13 +853,33 @@ def var(self, ddof=1, *args, **kwargs):
 
     @Appender(GroupBy.size.__doc__)
     def size(self):
-        # It's a special case as higher level does return
-        # a copy of 0-len objects. GH14962
         result = self._downsample("size")
-        if not len(self.ax) and isinstance(self._selected_obj, ABCDataFrame):
+        if not len(self.ax):
             from pandas import Series
 
-            result = Series([], index=result.index, dtype="int64")
+            if self._selected_obj.ndim == 1:
+                name = self._selected_obj.name
+                result = self._selected_obj.__class__(
+                    [], index=result.index, dtype="int64", name=name
+                )
+            else:
+                result = Series([], index=result.index, dtype="int64")
+        return result
+
```

Comment:
is it possible to move this code change to make _downsample handle this better? I am harping on this because this is very duplicative (not your fault), but still want to avoid this if at all possible.

Reply:
The proper way do that, would be to remove all those "empty length" special cases. I tried to do that, removing this snippet of `DatetimeIndexResampler._downsample()`:

```python
if not len(ax):
    # reset to the new freq
    obj = obj.copy()
    obj.index.freq = self.freq
    return obj
```

but this then triggers other empty-data problems down the stack (such as the block manager layer which I currently don't really understand).

I'll try to look at it further.